Nrf52 nffs example

vikrant8052 opened the pull request.

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   CMakeLists.txt
	new file:   samples/boards/nrf52/nffs/CMakeLists.txt
	new file:   samples/boards/nrf52/nffs/README.rst
	new file:   samples/boards/nrf52/nffs/prj.conf
	new file:   samples/boards/nrf52/nffs/sample.yaml
	new file:   samples/boards/nrf52/nffs/src/main.c

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

dbkinder reviewed (changes requested):
Doc needs to follow sample doc guidelines please.

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

vikrant8051 commented:
I've update my Readme.rst as per your suggestion. Please consider it.

vikrant8051 commented:
I've updated my Readme.rst as per your suggestion. Please consider it.
Thanks !!

On Sun, Feb 11, 2018 at 12:17 AM, David Kinder <notifications@github.com>
wrote:

> *@dbkinder* requested changes on this pull request.
>
> Doc needs to follow sample doc guidelines please.
> ------------------------------
>
> In samples/boards/nrf52/nffs/README.rst
> <https://github.com/zephyrproject-rtos/zephyr/pull/6123#discussion_r167407530>
> :
>
> > @@ -0,0 +1,48 @@
> +To successfully run this demo example,
>
> This document needs to be in ReST format, so please follow the sample doc
> template found in doc/templates/sample.tmpl or start with an existing
> sample doc such as samples/boards/nrf52/mesh/onoff-app/README.rst and
> edit for your sample.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/6123#pullrequestreview-95630026>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFs-RUY1qUws2GipdzHU7FmWFSYPwVo6ks5tTeRagaJpZM4SBAvG>
> .
>

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

dbkinder commented on samples/boards/nrf52/nffs/README.rst:
```diff
@@ -1,11 +1,30 @@
+Building and Running
+********************
+
+To build this example, we have to add "ext/fs/nffs/include" into list of
```
How about this instead:

    To build this example, edit the `$ZEPHYR_BASE/CMakeLists.txt`` file to add ``ext/fs/nffs/include``
    to the include directories listed there, so it will look like this:

    .. code-block:: none

      zephyr_include_directories(
        ${STDINCLUDE}
        ext/fs/nffs/include
      )

    After this modification you can build the sample using these commands:

    .. code-block:: bash

       cd $(ZEPHYR_BASE)/samples/boards/nrf52/nffs
       mkdir build
       cd build
       cmake -DBOARD=nrf52840_pca10056 ..
       make
       cd zephyr
       make clean
       make all
 
    After the build completes, the resulting binary ``zephyr.hex`` will be
    in the current directory.  Use this file to flash the board using the
    ``nrfjprog`` utility.

    After flashing to the nrf52840_pdk board, open a serial terminal to see console messages
    written by the application:

    - Press Button 2 on the PDK board to create and write data in file ``0.txt``
    - Press Button 1 to read data from ``0.txt``

    For more information about the Zephyr file system, see the :ref:`file_system` documentation. 

I'm not sure about the build instructions you've written here (I've just made them formatted nicely).  I'd recommend one of the development engineers (@SebastianBoe perhaps) review these build instructions, including how to get the include added (I think you're supposed to create a CMakeLists.txt local to the sample rather than editing the one in ZEPHYR_ROOT.)

dbkinder reviewed (changes requested):
Thanks, that's closer, but still needs formatting and wording fixes. @SebastianBoe can you verify the build instructions (they look wonky to me).

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   ../../../CMakeLists.txt
	modified:   nffs/CMakeLists.txt
	modified:   nffs/README.rst

vikrant8051 commented:
Thanks I've edited project's README.rst as per your suggestions. 

Plus editing of touching $(zephyr_base)/CMAkeLists.txt, I edit project's own CMakeLists.txt to include 
ext/fs/nffs/include directory.

Please consider it.

Thank You !!

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   nffs/README.rst

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

dbkinder reviewed (changes requested):
getting closer.  I'd still like a technical review by one of the project developers (I'm a tech writer with the project).

vikrant8052 committed:
Update README.rst

vikrant8051 commented on samples/boards/nrf52/nffs/README.rst:
```diff
@@ -0,0 +1,48 @@
+This sample can be found under ``$(zephyr_base)/samples/boards/nrf52/nffs`` in the
+Zephyr tree.
+
+I've already included ``$(zephyr_base)/ext/fs/nffs`` directory 
```
Sorry for my English. It's my first time. Thanks for correcting me.
As per your suggestions, I've edited README.rst. Please have a look.

Thank You !!

nvlsianpu commented:
Thanks for your efforts! It is really appreciated by us.

Can you move this example to destination: `samples/subsys/fs/nffs`? I think it is not really nRF5x specific - someone might want to adapt it to another target.

Squash all commits to one (you can use for instance `git rebase -i HEAD~17`).
Align the commit message to zephyr pattern. See [the gide](http://docs.zephyrproject.org/contribute/contribute_guidelines.html) or look at dferentt commit like [this one](https://github.com/zephyrproject-rtos/zephyr/commit/219b3cd06712377c83c5ae3486effe888836a518#diff-9c332a0b6107a18bd34bca0c6952439b)

nvlsianpu commented on samples/boards/nrf52/nffs/CMakeLists.txt:
```diff
@@ -0,0 +1,12 @@
+
+
+zephyr_library_include_directories(
+	$ENV{ZEPHYR_BASE}/samples/bluetooth
```
I don't understand why you need to include any bluetooth code here.

vikrant8051 replied:
I've removed it.

nvlsianpu commented on samples/boards/nrf52/nffs/sample.yaml:
```diff
@@ -0,0 +1,8 @@
+tests:
+  test:
+    build_only: true
+    platform_whitelist: qemu_cortex_m3 qemu_x86
```
platform_whitelist: nrf52840_pca10056 nrf52_pca10040

In case you want to have quemu as well here (which is good Idea) you need to provide another proj.conf file for it, (like this one: https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/subsys/fs/nffs_fs_api/prj.conf) and  ram beckend: https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/subsys/fs/nffs_fs_api/src/test_ram_backend.c) - I gues you won't do that.

nvlsianpu commented on samples/boards/nrf52/nffs/sample.yaml:
```diff
@@ -0,0 +1,8 @@
+  test:
+    build_only: true
+    platform_whitelist: qemu_cortex_m3 qemu_x86
+    tags: bluetooth
```
tags: file system
@galak ^^

nvlsianpu commented on samples/boards/nrf52/nffs/sample.yaml:
```diff
@@ -0,0 +1,8 @@
+sample:
+  description: TBD
```
sth like: Example for show NFFS usage.

nvlsianpu commented on samples/boards/nrf52/nffs/sample.yaml:
```diff
@@ -0,0 +1,8 @@
+sample:
+  description: TBD
+  name: TBD
```
NFFS file operations example.

nvlsianpu commented on samples/boards/nrf52/nffs/src/main.c:
```diff
@@ -0,0 +1,186 @@
+	while(lock==0)
+	{
+
+        	if(CHECKB(NRF_P0->IN,12)==0)
```
generic gpio driver api can be use here 'gpio_read'

vikrant8051 replied:
removed it.

nvlsianpu commented on samples/boards/nrf52/nffs/src/main.c:
```diff
@@ -0,0 +1,186 @@
+void main(void)
+{
+
+	NRF_P0->PIN_CNF[12]=0x0000000C;
```
Why gpio API is not used here?

nvlsianpu commented on samples/boards/nrf52/nffs/README.rst:
```diff
@@ -0,0 +1,47 @@
+   mkdir build
+   cd build
+   cmake -DBOARD=nrf52840_pca10056 ..
+   make
```
normally an example build is done after 1st make commend. @SebastianBoe why addytional command is needed here? Can you suggest improvement for reducing this to "one make"?

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/CMakeLists.txt
	modified:   samples/boards/nrf52/nffs/README.rst
	modified:   samples/boards/nrf52/nffs/src/main.c

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	renamed:    samples/boards/nrf52/nffs/CMakeLists.txt -> samples/subsys/fs/nffs/CMakeLists.txt
	renamed:    samples/boards/nrf52/nffs/README.rst -> samples/subsys/fs/nffs/README.rst
	renamed:    samples/boards/nrf52/nffs/prj.conf -> samples/subsys/fs/nffs/prj.conf
	renamed:    samples/boards/nrf52/nffs/sample.yaml -> samples/subsys/fs/nffs/sample.yaml
	renamed:    samples/boards/nrf52/nffs/src/main.c -> samples/subsys/fs/nffs/src/main.c

vikrant8052 committed:
Merge branch 'nrf52_nffs_example' of https://github.com/vikrant8051/zephyr into nrf52_nffs_example

Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

vikrant8052 committed:
Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	modified:   samples/boards/nrf52/nffs/README.rst

Signed-off-by: vikrant8051 <5979717+vikrant8051@users.noreply.github.com>

	new file:   samples/boards/nrf52/nffs/README.rst
